prow/jira/jira.go

```diff
@@ -121,9 +134,27 @@ func (u userAgentSettingTransport) RoundTrip(r *http.Request) (*http.Response, e
 	return u.upstream.RoundTrip(r)
 }
 
+type clientUsedTransport struct {
+	used     bool
+	upstream http.RoundTripper
+}
+
+func (c clientUsedTransport) RoundTrip(r *http.Request) (*http.Response, error) {
```

Comment:
This needs to be threadsafe - It happens to not matter today since there is only one plugin that uses this and we reconstruct per event, but that might be different in the future

Reply:
Sorry, I don't get it. Why threadsafe? Every time ForPlugin is called, a completely new transport along with Jira client is created. Maybe missed something...